web/src/main/java/com/bazaarvoice/emodb/web/scanner/rangescan/LocalRangeScanUploader.java

```diff
@@ -255,8 +266,17 @@ public void run() {
             int partCountForFirstShard = 1;
             Batch batch = new Batch(context, partCountForFirstShard);
 
+            // check if there is a stash cut off time.
+            Map<String, StashRunTimeInfo> stashTimeInfoMap = _compactionControlSource.getStashTimesForPlacement(placement);
+            DateTime cutoffTime = null;
+            if (stashTimeInfoMap.size() > 0) {
+                cutoffTime = new DateTime(stashTimeInfoMap.entrySet().stream()
+                        .min((entry1, entry2) -> entry1.getValue().getTimestamp() > entry2.getValue().getTimestamp() ? 1 : -1)
+                        .get().getValue().getTimestamp());
+            }
```

Comment:
There's a slight problem here in that if 2 stashes are running at the same time but started at different times this will always use the cutoff for the earlier stash in both stashes.  This probably isn't important since we don't run more than one stash concurrently, but it might be cleaner if there were a way to get the cutoff which corresponds with this stash's ID.  I'm ok leaving it, but something to consider.